### PR TITLE
Feature/team members

### LINKED ISSUE #41: projects/events: add team members
The projects and the events (the "Item" general class created in #23) should contain teams of their current members or organizers. We should:
- [x] Add a list of members to the Item's model
- [x] Create endpoints capable of adding and removing team members

### PR DESCRIPTION
**Pull request by MRita443**
Closes #41

Adds a list of team members to Activity's model and includes endpoints for adding and removing team members

# Review checklist
-   [ ] Properly documents API changes in `docs/openapi.yml`
-   [x] Contains enough appropriate tests
-   [x] Behavior is as expected
-   [x] Clean, well structured code

### CODE REVIEW

File: src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/EventDto.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/EventDto.kt]:
```suggestion
    val teamMembersIds: List<Long>?
```
I didn't check my self but I think this works, right? It's best to not wonder why it's mutable

File: src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/ProjectDto.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/ProjectDto.kt]: same as above

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/AccountService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/AccountService.kt]:
```suggestion
    fun doesAccountExist(id: Long): Boolean = repository.findByIdOrNull(id) != null
```
I think something like this is more explicit, wdyt?

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt]: Is this annotation supposed to be here? I think it's used for entities

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt]: dead code (should be picked up by the linter anyway)

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt:28]:
```suggestion
            ErrorMessages.accountNotFound(idAccount)
```
small detail to reduce clutter

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/EventService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/EventService.kt:16]: Isn't the team members' logic missing when creating events? We should test that too. I think the tests exist for the projects' version

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ProjectService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ProjectService.kt]:
```suggestion
       dto.teamMembersIds?.forEach {
            val account = accountService.getAccountById(it)
            project.teamMembers.add(account)
        }
```
Using forEach avoids the need for `orEmpty()`, the function won't execute if the collection is null

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ProjectService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ProjectService.kt]: Same comment with forEach. You can also try exploring the `apply { }` function to combine clear and forEach if you'd like

File: src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt
Comment (BrunoRosendo) [inline src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt]: Why did this disappear? Also, we should probably test if teamMembers are returned

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]
Good job! This feature will be very important going forward :rocket: I left some cosmetic and minor comments

Also, you should fix the linting (formatting) issues, please check README for details. We usually have an automatic action running in the PR but it doesn't trigger because you're merging into another branch. It'll appear once that branch is merged to develop

File: src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/EventDto.kt
Comment (MRita443) [reply src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/EventDto.kt]: Yes, I had it setup like this from when I was still attempting to use the same MutableList<Account> for the dtos, changed now

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/EventService.kt
Comment (MRita443) [reply src/main/kotlin/pt/up/fe/ni/website/backend/service/EventService.kt:16]: For sure, it was an oversight and should be working and being tested properly now

File: src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt
Comment (MRita443) [reply src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt]: It was an accident, probably removed when deleting some other code, it's back now!

**Review by BrunoRosendo — APPROVED** [review]
Looks good, congrats! Let's just wait for the other PRs to get merged :(

Also, whoever merged first between this PR and #55 needs to address the teamMembers change and respective tests

**Comment by BrunoRosendo** [comment]
@MRita443 You should update your branch with the latest changes. Then we just need another review and it's good to merge

File: src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt
Comment (bdmendes) [inline src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt]: This is a bit awkward. What do you think of making `ProjectService` extend `ActitvityService`?

**Review by bdmendes — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt
Comment (MRita443) [reply src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt]: Tried to implement this on my latest commit, wasn't sure of a good way to tie in the `ActivityRepository` to `ProjectService` and `EventService`, let me know what you think

Comment (bdmendes) [reply src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt]: The repository should not be tied to the service. I think this structure should be possible:
```
ProjectService(projectRepository) : ActivityService(projectRepository)
```
Since `ProjectRepository` is also an `ActivityRepository`.

Comment (BrunoRosendo) [reply src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt]: I haven't tested anything but this structure might work for what you're suggesting: https://stackoverflow.com/questions/43350397/spring-data-jpa-inheritance-in-repositories

Comment (MRita443) [reply src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt]: The inheritance between repositories was added in my latest commits, check it out!

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt
Comment (BrunoRosendo) [inline src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt]: this is kind of weird because, when using a ProjectService, we'll have `getActivityById` and `getProjectById`. Furthermore, activity one will return an error related to the activity. Maybe we could set this function as private and it won't appear in project and event

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt
Comment (MRita443) [reply src/main/kotlin/pt/up/fe/ni/website/backend/service/ActivityService.kt]: I agree, just modified it

File: gradle/wrapper/gradle-wrapper.properties
Comment (BrunoRosendo) [inline gradle/wrapper/gradle-wrapper.properties:3]: not sure why this changed but it's ok

**Review by BrunoRosendo — APPROVED** [review]
Great job!

File: gradle/wrapper/gradle-wrapper.properties
Comment (MRita443) [reply gradle/wrapper/gradle-wrapper.properties:3]: Hm, I recall I did upgrade Gradle whilst working on this, but don't recall manually changing this at all

Comment (BrunoRosendo) [reply gradle/wrapper/gradle-wrapper.properties:3]: It was probably automatic, but is should be fine to upgrade gradle

**Comment by BrunoRosendo** [comment]
@MRita443 You were the unlucky one, this PR needs a rebase and you should also change the "event update" endpoint to reflect the team changes, just like the projects

**Review by BrunoRosendo — APPROVED** [review]
Approving after merge :-)

**Review by Sirze01 — APPROVED** [review]
Good job!

**Review by bdmendes — APPROVED** [review]
Good work!

**Pull request merged**
